### PR TITLE
Add support for pseudoClassEnabledElements in percySnapshot options

### PR DESCRIPTION
**Pull request by Manoj-Katta**
## Description

This PR adds support for the `pseudoClassEnabledElements` option in `percySnapshot` to allow capturing snapshots with pseudo-class states enabled on specific elements.

## Changes

- **Added** `pseudoClassEnabledElements` option support in `percySnapshot` function
- The option can be passed directly to `percySnapshot` or configured globally via Percy config
- When passed directly, it takes precedence over the global config setting
- Falls back to `utils.percy.config.snapshot.pseudoClassEnabledElements` when not explicitly provided

## Implementation Details

The implementation follows a two-tier configuration approach:
1. Check if `pseudoClassEnabledElements` is passed in the `options` parameter
2. If not provided, fall back to the global Percy config setting
3. Pass the resolved value to the snapshot request

## Testing

Added comprehensive test coverage:
- ✅ Test that global Percy config `pseudoClassEnabledElements` is used when option is not passed
- ✅ Test that explicit `pseudoClassEnabledElements` option takes precedence over global config

### CODE REVIEW

File: addon-test-support/@percy/ember/index.js [hunk 1]
```diff
@@ -66,6 +66,13 @@ export default async function percySnapshot(name, {
       eval(await utils.fetchPercyDOM());
     }
 
+    const pseudoClassEnabledElements = options.pseudoClassEnabledElements ||
+    utils.percy?.config?.snapshot?.pseudoClassEnabledElements;
+
+    if (pseudoClassEnabledElements) {
```
[the remark this answers is not in the document]
Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: The user won't pass null in the pseudoClassEnabledElements if he passed  then it would be better to capture the elements present in config file

File: addon-test-support/@percy/ember/index.js [hunk 2]
```diff
@@ -66,6 +66,13 @@ export default async function percySnapshot(name, {
       eval(await utils.fetchPercyDOM());
     }
 
+    const pseudoClassEnabledElements = options.pseudoClassEnabledElements ||
+    utils.percy?.config?.snapshot?.pseudoClassEnabledElements;
```
Comment (rahulXbrowserstack) [inline addon-test-support/@percy/ember/index.js]: If at the snapshot level I keep the `pseudoClassEnabledElements` as false, but lets say its still set at global level lets say `{ selector: ['.my-class'] }`, then `false || { selector: ['.my-class'] }` will be picking up that selector for that snapshot, but that shouldn't happen right ?
since we written false at snapshot level, shouldn't that be considered ??
or will it not be boolean ? it will always be array of string ? did we checked the type checking is working or not ?

File: addon-test-support/@percy/ember/index.js [hunk 3]
```diff
@@ -66,6 +66,13 @@ export default async function percySnapshot(name, {
...
+
+    if (pseudoClassEnabledElements) {
+      options = { ...options, pseudoClassEnabledElements };
```
Comment (rahulXbrowserstack) [inline addon-test-support/@percy/ember/index.js]: also, if it's already in options, doing the spread options = { ...options, ... } looks like a bit redundant. Why are we doing like this ?

File: tests/acceptance/index-test.js [hunk 4]
```diff
@@ -116,6 +125,39 @@ module('percySnapshot', hooks => {
...
+        pseudoClassEnabledElements: {
+          selector: ['#ember-testing']
+        }
+      });
+
+      let reqs = await helpers.get('requests');
+      let snapshotReq = reqs.filter(req => req.url === '/percy/snapshot').pop();
+      assert.deepEqual(snapshotReq.body.pseudoClassEnabledElements, {
+        selector: ['#ember-testing']
+      });
```
Comment (rahulXbrowserstack) [inline tests/acceptance/index-test.js:159]: if you apply the code tweak above, it would be good to add one more quick test case to prove the edge case works. something like:
"allows overriding and disabling global pseudoClassEnabledElements with explicit false/null"

just set a global config, pass pseudoClassEnabledElements: false into the percySnapshot options, and assert that the req body actually receives false (or doesn't include the global config).

**Review by rahulXbrowserstack — COMMENTED** [review]
@Manoj-Katta  Can you also attach some screenshots to show, the dev testing done

@rahulXbrowserstack Since I can't attach the staging builds I have written the tests that I have done in the jira ticket itself

File: addon-test-support/@percy/ember/index.js (hunk 2 above)
Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: In the pseudoClassEnabledElements we will have either id: [], className: [], selector: [], xPath:[], but not a boolean value like false, so thats why I have removed that

Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: Ref: https://github.com/percy/cli/pull/2141/changes#diff-e7c1269ddb64f1de14c2c79cac204c16cfc3f8d8535512d218f8e4ad8efc63b6R287

Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: if(!options.pseudoClassEnabledElements && utils.percy?.config?.snapshot?.pseudoClassEnabledElements){
      options = { ...options, pseudoClassEnabledElements: utils.percy.config.snapshot.pseudoClassEnabledElements };
    }
    
    so I am thinking to use something like this is that fine

Comment (rahulXbrowserstack) [reply addon-test-support/@percy/ember/index.js]: lets say there are 1000 snapshots, and we passed a global config, now that will be applied to all 1000 snapshots, but what if I want not to apply on just 1 specific snapshot, how will I pass the options ??

Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: so for this case, we need to first trigger the popover elements first in percy and upon that we can use :popover-open in selector which will open all the popovers that we need but not the one since we haven't opened that in test script itself

File: tests/acceptance/index-test.js (hunk 4 above)
Comment (Manoj-Katta) [reply tests/acceptance/index-test.js:159]: pseudoClassEnabledElements can't be false/null

File: addon-test-support/@percy/ember/index.js (hunk 3 above)
Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: Removed the redundant options adding

File: addon-test-support/@percy/ember/index.js [hunk 5]
```diff
@@ -66,6 +66,10 @@ export default async function percySnapshot(name, {
       eval(await utils.fetchPercyDOM());
     }
 
+    if (!options.pseudoClassEnabledElements && utils.percy?.config?.snapshot?.pseudoClassEnabledElements) {
```
Comment (amandeepsingh333) [inline addon-test-support/@percy/ember/index.js]: any reason for doing this inside this fxn
and create this to a helper function to update options

Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: The psuedoClassEnabledElements are passed in options of the percySNapshot command, the pseudoClassENabledElements can also be passed via .percy.yml file and if the pseudoClassEnabledELements is not present in options of percySnapshot command then it need to take from the yml file thats why its added here

Comment (Manoj-Katta) [reply addon-test-support/@percy/ember/index.js]: Also changed this to a helper function

**Review by amandeepsingh333 — APPROVED** [review]

**Review by rahulXbrowserstack — APPROVED** [review]
